### PR TITLE
Simplify SafeArea test for maintainBottomViewPadding to ensure maintainBottomViewPadding is always respected

### LINKED ISSUE #97609: [IOS] SafeArea maintainBottomViewPadding is not respected when keyboard viewInsets are animated
This issue has been introduced as a result of the changes to the flutter engine (iOS) made in PR 29281 (https://github.com/flutter/engine/pull/29281). Those changes are in use on the current master and dev branch of this project. 

It is now possible for MediaqQuery viewInsets.bottom to be >0 AND for padding.bottom >0 during the keyboard transition. This results in the maintainBottomViewPadding property being ignored within SafeArea build() at various times during the animation.

## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Tap on the TextField to reveal the software keyboard
3. Tap 'done' to remove the software keyboard

**Expected results:** 
The keyboard should overlay the app and not affect the layout of any widgets. In particular, the SafeArea padding at the bottom of the orange widget should remain unchanged regardless of the viewInsets.bottom value created by the keyboard  since maintainBottomViewPadding is set to true.

**Actual results:** 
During animation, the orange widget 'jumps' downward. It is laid out differently during the animation indicating that the bottom viewPadding is not always respected, despite maintainBottomViewPadding being set to true.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      resizeToAvoidBottomInset: false,
      appBar: AppBar(
        title: const Text('SafeArea Demo'),
      ),
      body: Center(
        child: Column(
          children: <Widget>[
            const Expanded(
                child: Padding(
              padding: EdgeInsets.only(top: 20.0),
              child: Text("This scaffold has resizeToAvoidBottomInset=false"),
            )),
            SafeArea(
              maintainBottomViewPadding: true,
              child: Container(
                color: const Color.fromARGB(255, 255, 72, 0),
                height: 500,
                child: Column(
                  children: const [
                    Text(
                      'This container is surrounded by a SafeArea \n with maintainBottomViewPadding=true',
                      style: TextStyle(fontWeight: FontWeight.bold),
                    ),
                    SizedBox(width: 200, child: TextField()),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
No relevant runtime logs
```

```
No issues found! (ran in 1.8s)
```

```
[✓] Flutter (Channel master, 2.10.0-1.0.pre.373, on macOS 12.1 21C52 darwin-x64, locale en-GB)
    • Flutter version 2.10.0-1.0.pre.373 at /Users/joellurcook/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a22cf5d6b9 (28 minutes ago), 2022-02-02 09:16:05 +0100
    • Engine revision 0bd729b972
    • Dart version 2.17.0 (build 2.17.0-74.0.dev)
    • DevTools version 2.10.0

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/joellurcook/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio 2.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Community Edition (version 2021.1.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.63.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.34.0

[✓] Connected device (2 available)
    • iPhone 13 (mobile) • 63C9A42F-6841-4F83-8922-DCAE0C900EBE • ios            • com.apple.CoreSimulator.SimRuntime.iOS-15-2 (simulator)
    • Chrome (web)       • chrome                               • web-javascript • Google Chrome 97.0.4692.99

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

https://user-images.githubusercontent.com/55881597/152073161-6c91ba08-fa50-4b38-a48c-6961bc95fdf7.mp4

### PR DESCRIPTION
**Pull request by joellurcook**
Recent flutter engine changes enable iOS MediaQuery bottom viewInsets and padding to be !=0 simultaneously (see https://github.com/flutter/engine/pull/29281). 
This condition resulted in SafeArea maintainBottomViewPadding being ignored during keyboard animation.
This commit forces use of viewPadding.bottom when maintainBottomViewPadding is true.

|Before|After|
|---|---|
|https://user-images.githubusercontent.com/55881597/152152622-1f9c8400-ca6b-4ea5-b58c-cff7c7b806d9.mp4|https://user-images.githubusercontent.com/55881597/152152697-75ecfeaf-1ea5-4eb0-a224-f4cefa8744d8.mp4|

This fixes the following issue:
fixes #97609.

No changes to existing tests.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by joellurcook** [comment]
I offer the following justification for the much-simplified condition concerning bottom padding:

Assuming maintainBottomViewPadding is true and knowing that padding is calculated...

- If viewInsets >= viewPadding (padding = 0), we should use viewPadding (this was the case previously)
- If  0 < viewInsets < viewPadding (padding = viewPadding - viewInsets), we should use viewPadding (this represents animating keyboard, and fixes the graphical glitch) 
- If viewInsets == 0 (padding = viewPadding), there is no difference between padding and viewPadding

In all cases, it is valid to use viewPadding when maintainBottomViewPadding is true. This is also semantically aligned with the property name itself, which does not imply conditional behaviour based on viewInsets/padding.

File: packages/flutter/test/widgets/safe_area_test.dart [hunk 1]
```diff
@@ -184,6 +218,44 @@ void main() {
         final Offset finalPoint = tester.getCenter(find.byType(Placeholder));
         expect(initialPoint, finalPoint);
       });
+
+      testWidgets(
+          'SafeArea with nested Scaffold  - partial ViewInsets consume Padding',
+          (WidgetTester tester) async {
```
Comment (Piinks) [inline packages/flutter/test/widgets/safe_area_test.dart:224]: Style guide nit: These should all be one line

File: packages/flutter/test/widgets/safe_area_test.dart [hunk 2]
```diff
@@ -147,6 +147,40 @@ void main() {
         expect(initialPoint, finalPoint);
       });
 
+      testWidgets('SafeArea alone - partial ViewInsets consume Padding',
+          (WidgetTester tester) async {
```
Comment (Piinks) [inline packages/flutter/test/widgets/safe_area_test.dart:151]: Same here

**Review by Piinks — COMMENTED** [review]
Hey @joellurcook, thanks for the contribution! Just a few nits below per the style guide. Thanks for fixing this!

File: packages/flutter/test/widgets/safe_area_test.dart (hunk 1 above)
Comment (joellurcook) [reply packages/flutter/test/widgets/safe_area_test.dart]: Fixed in commit c82919c

File: packages/flutter/test/widgets/safe_area_test.dart (hunk 2 above)
Comment (joellurcook) [reply packages/flutter/test/widgets/safe_area_test.dart]: Fixed in commit c82919c

**Comment by joellurcook** [comment]
I assume the failing checks are unrelated to the PR content - is there a simple way to prod the checks to be performed again?

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

The failing checks are unrelated to your PR, if you rebase to update with the latest from the master branch, they should pass. Thanks again!

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**